proto/frequenz/api/reporting/v1/reporting.proto

```diff
@@ -0,0 +1,534 @@
+//
+// !!! note
+//     Specifies which additional fields should be included in the response.
+//
+message IncludeOptions {
+  // Defines whether to include results in the response message.
+  enum FilterOption {
+    FILTER_OPTION_UNSPECIFIED = 0;
+    FILTER_OPTION_EXCLUDE = 1;
+    FILTER_OPTION_INCLUDE = 2;
+  }
+
+  // Optional bound inclusion. By default, bounds are not included in the response.
+  optional FilterOption bounds = 1;
+
+  // Optional operational state inclusion. By default, states are not included in the response.
+  optional FilterOption states = 2;
+}
+
+// Message defining the aggregation configuration for a custom formula within a specific microgrid.
+//
+// The AggregationConfig allows clients to specify how metrics should be aggregated across
+// microgrid components. It serves as the configuration guideline for any aggregation operation.
+//
+// !!! example
+//    To calculate the total voltage across three components in series with IDs 1, 2, and 3, the
+//     configuration could look like:
+//     {
+//       microgrid_id: 42,
+//       metric: DC_VOLTAGE_V,
+//       aggregation_formula: "#1 + #2 + #3"
+//     }
+//
+message AggregationConfig {
+  // ID of the microgrid for which the formula is being specified.
+  uint64 microgrid_id = 1;
+
+  // The metric that is to be aggregated using the specified formula.
+  //
+  // !!! caution
+  //     Ensure the chosen metric is supported by all relevant microgrid components.
+  //     Failure to meet these conditions might results in an aggregation error.
+  frequenz.api.common.v1.metrics.Metric metric = 2;
+
+  // The formula used for aggregating the component metric for this microgrid.
+  //
+  // !!! info
+  //     Two types of aggregation formulas are supported:
+  //     1. Aggregate functions: These are standard aggregation functions like `sum()` and `avg()`.
+  //        Example: `sum(#1,#2)`, `avg(#3,#4)`
+  //
+  //     2. Dedicated formulas: These support basic math operators while concatenating microgrid
+  //        component IDs.
+  //        Example: `#1 + #2 - #3`, `(#3 * #2) /# 1`
+  string aggregation_formula = 3;
+}
+
+// Defines a single aggregated metric sample, which represents a consolidated metric value
+// over a collection of microgrid components within a specific timeframe.
+//
+// This message serves to encapsulate aggregate metrics derived from multiple component
+// measurements. It is particularly useful in scenarios where an overall metric needs to be
+// calculated for a set of components. For instance, you could use it to represent the average
+// voltage across multiple sources within a microgrid or to calculate the average state of
+// charge of several batteries.
+//
+message SimpleAggregatedMetricSample {
+  // The UTC sample timestamp for the aggregated metrics.
+  //
+  // !!! info
+  //     This timestamp indicates the specific moment when the aggregated metric value
+  //      is calculated.
```

Comment:
Good question. I would suggest opening an issue for this, as this will be an important implementation detail. Perhaps the sampled_at field can be omitted if it is not useful?

Reply:
Honestly I think this documentation is just wrong. Time series where time information is lost is useless. I suggest to fix this doc now.

Reply:
What would you suggest to change it to? Which timestamp should this field represent, if not the moment of calculation (which, as you suggested, will make all the timestamps roughly the same as when the request was sent).

Reply:
The original timestamp of the samples that were aggregated, i.e. the (potentially resampled) measurement timestamp. These should all be the same, which touches my other comment about always requiring resampling when aggregating components.

Reply:
Makes sense.